Improve documentation of LuigiMockModule (#2105)
SAP/luigi: client-frameworks-support/testing-utilities/README.md

> ## Luigi Testing Utilities

Reviewer: ```suggestion
# Luigi Testing Utilities
```

> Luigi Testing Utilities is a set of auxiliary functions used to enhance the user experience while testing Luigi based micro frontends. The functions abstract away Luigi specific logic from the tester so that it is easier for them to mock and assert Luigi functionality. 

Reviewer: ```suggestion
The Luigi Testing Utilities are a set of auxiliary functions used to enhance the user experience while testing Luigi-based micro frontends. The functions abstract away Luigi-specific logic from the tester so that it is easier for them to mock and assert Luigi functionality. 
```

> ### LuigiMockUtil 

Reviewer: ```suggestion
## LuigiMockUtil 
```

This class contains certain utility helper functions needed when writing [protractor](https://www.npmjs.com/package/protractor) based e2e tests. You can simply import this module into you project and then use an instance of it to test microfrontend functionality. 

> ### How to use the library

Reviewer: ```suggestion
### How to use the library
```

I'm not sure if "the library" means the utilities as a whole or only the LuigiMockUtil class? Because we don't mention a library before this title and it's a little confusing.

Reviewer 2: I think it is fine, because we also use the term in the left nav of our docu as well (Framework support library).

> **Prerequisites:**

Reviewer: ```suggestion
### Prerequisites
```

      someOtherData: 'randomInfo',
    });
  }
}
```

#### Functions provided

> - **mockContext**: Mocks the context by sending luigi context messegaes with the desired mocked context as parameter. 

Reviewer: ```suggestion
- **mockContext**: Mocks the context by sending Luigi context messages with the desired mocked context as parameter. 
```

> - **mockPathExists**: This method serves as a mock for the luigi client pathExists() function. It is used in e2e tests when component being tested utilizes a call to `LuigiClient.linkManager().pathExists()`

Reviewer: ```suggestion
- **mockPathExists**: This method serves as a mock for the Luigi Client `pathExists()` function. It is used in e2e tests when component being tested utilizes a call to `LuigiClient.linkManager().pathExists()`
```

> - **modalOpenedWithTitle**: Checks on the printed DOM Luigi message responses for a modal with given title having been opened. In such a case a message would be printed containing a modal.title. Returns false if not such element was found.

Reviewer: ```suggestion
- **modalOpenedWithTitle**: Checks on the printed DOM Luigi message responses for a modal with given title being opened. In such a case, a message would be printed containing a `modal.title`. Returns `false` if such element was not found.
```

- **getMSG**: Return list of messages, representing message elements added in the DOM for testing. 

> - **parseLuigiMockedMessages**: Parses the elements added by LuigiMockModule into the DOM and assigns them to the local messages variable

Reviewer: ```suggestion
- **parseLuigiMockedMessages**: Parses the elements added by LuigiMockModule into the DOM and assigns them to the local messages variable.
```
